doc/source/profiling.md

```diff
@@ -0,0 +1,76 @@
+# Profiling
+
+xDEM has a built-in profiling tool, that can be used to provide more insight on the memory and time use of function.
+```{warning}
+Make sure to install xDEM with development dependencies included.
+```
```

Comment:
We should explain to the users the underlying tools explicitly: "The profiling functionalities rely on `psutil` and `plotly`, which can be installed manualyl or using xDEM development dependencies."
I guess we'll refine the "development dependencies" part once we've addressed #755!